src/coreComponents/mesh/generators/VTKMeshGeneratorTools.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file VTKMeshGeneratorTools.hpp
+ */
+
+#include "VTKMeshGeneratorTools.hpp"
+
+#include <vtkAppendFilter.h>
+#include <vtkDIYUtilities.h>
+#include <vtkDIYGhostUtilities.h>
+
+// NOTE: do NOT include anything from GEOSX here.
+// See full explanation in VTKMeshGeneratorTools.hpp.
+
+namespace geosx
+{
+namespace vtk
+{
+
+vtkSmartPointer< vtkUnstructuredGrid >
+redistribute( vtkPartitionedDataSet & localParts,
+              MPI_Comm mpiComm )
+{
+  // The code below is modified from vtkDIYKdTreeUtilities::Exchange():
```

Comment:
Why don't you use the original code?

Reply:
The VTK code is a bit more general and complex than needed here. For example, it takes an additional `assigner` parameter which specifies the distribution of partitions. Our use case is simpler, since every local partition goes directly to only one target rank, and respectively every rank receives exactly one partition from every other. Therefore we can simplify some data structures used in exchange (e.g. use `vector` instead of `vector<vector>` for storing blocks) and do strictly less work, even if it's just O(1) less.

Reply:
But could we call the function from the lib directly, even if we have to set up some info a little more complex?

Reply:
Ah, I see what you mean. No, we can't call it directly (that would've been my choice too) because it's only exposed in a header `vtkDIYKdTreeUtilities.h` that VTK considers "private" and does not install.

https://gitlab.kitware.com/vtk/vtk/-/blob/master/Filters/ParallelDIY2/CMakeLists.txt#L37